src/middlewared/middlewared/alert/source/m_series_nvdimm_firmware_freebsd.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding=utf-8 -*-
+import logging
+
+from middlewared.alert.base import Alert, AlertClass, AlertCategory, AlertLevel, AlertSource
+from middlewared.service_exception import CallError
+
+logger = logging.getLogger(__name__)
+
+
+class TrueNASMNVDIMMFirmwareVersionAlertClass(AlertClass):
+    category = AlertCategory.HARDWARE
+    level = AlertLevel.EMERGENCY
+    title = "Invalid NVDIMM Firmware Version"
+    text = (
+        "Your NVDIMM controller %(index)d is using an invalid firmware version %(version)s which can cause data loss. "
```

Comment:
The text needs to be changed. NVDIMM controller is confusing since there is an A controller and B controller. We'll probably need to specify which controller has the incorrect NVDIMM firmware.

(i.e. NVDIMM device on controller (A/B) is using invalid firmware....)

Reply:
This refers to the device number in /dev. The A/B controller is displayed for each alert in the UI and in the E-Mail.

Reply:
`NVDIMM device (/dev/%(index)d is using...`

Reply:
Agreed, perhaps loose the /dev/ and just use nvdimm%(index)d